### PR TITLE
Add SetRichPresence API

### PR DESCRIPTION
**Pull request by corfe83**
This seems to do the trick on windows, in my testing. This API is needed for the game to make your in-game status visible to show on the friends list.

### CODE REVIEW

**Comment by hajimehoshi** [comment]
Could you implement this for Linux and macOS too? Thanks,

File: steamworks_windows.go [hunk 1]
```diff
@@ -142,6 +145,20 @@ func (s steamFriends) GetPersonaName() string {
...
+	defer func() {
+		C.free(unsafe.Pointer(keyString))
+		C.free(unsafe.Pointer(valueString))
+	}()
+
+	_, err := theDLL.call(flatAPI_ISteamFriends_SetRichPresence, uintptr(s), uintptr(unsafe.Pointer(keyString)), uintptr(unsafe.Pointer(valueString)))
+	if err != nil {
+		panic(err)
+	}
+}
```
Comment (hajimehoshi) [inline steamworks_windows.go:156]:
```go
v, err := theDLL.call(...)
if err != nil {
    panic(err)
}
return byte(v) != 0
```

File: steamworks_windows.go [hunk 2]
```diff
@@ -3,6 +3,9 @@
 
 package steamworks
 
+// #include <stdlib.h>
+import "C"
```
Comment (hajimehoshi) [inline steamworks_windows.go:7]: Don't use Cgo. Instead, we can create a C string pointer from a Go string by this:

```go
func goStringToCString(str string) (ptr unsafe.Pointer, free func()) {
    if !strings.HasSuffix(str, "\0") {
        str += "\0"
    }
    return unsafe.Pointer(&str[0]), func() { runtime.KeepAlive(str) }
}
```

Comment (hajimehoshi) [inline steamworks_windows.go]: Don't use Cgo. Instead, we can add this code:

```go
ckey := append([]byte(key), 0)
defer runtime.KeepAlive(ckey)
```

See `ClearAchievement` for example.

**Comment by corfe83** [comment]
> Could you implement this for Linux and macOS too? Thanks,

Yes, I mistakenly thought this library didn't implement those. I failed to notice the _unix.go file 😁 -- I will add support.

**Comment by corfe83** [comment]
Thanks for feedback, I've updated the PR now. I tested the unix code on MacOS only so far, but it should work for linux since the code is shared.

**Comment by corfe83** [comment]
I also fixed the return value, it was supposed to be bool

File: steamworks_unix.go [hunk 3]
```diff
@@ -303,6 +303,20 @@ func (s steamFriends) GetPersonaName() string {
 	return C.GoString(C.uintptrToChar(C.uintptr_t(v)))
 }
 
+func (s steamFriends) SetRichPresence(key, value string) bool {
+	ckey := append([]byte(key), 0)
+	defer runtime.KeepAlive(ckey)
```
Comment (hajimehoshi) [inline steamworks_unix.go]: Let's use C.CString and C.free for consistency for UNIX.

Comment (corfe83) [reply steamworks_unix.go]: Thanks, I've addressed this now and re-tested with latest push

**Review by hajimehoshi — APPROVED** [review]
LGTM, thanks!

**Pull request merged**